hummingbot/connector/exchange/huobi/huobi_api_order_book_data_source.py

```diff
@@ -114,17 +55,14 @@ async def get_snapshot(self, trading_pair: str) -> Dict[str, Any]:
         snapshot_data: Dict[str, Any] = await response.json()
         return snapshot_data
 
-    async def get_new_order_book(self, trading_pair: str) -> OrderBook:
-        snapshot: Dict[str, Any] = await self.get_snapshot(trading_pair)
-        timestamp = snapshot["tick"]["ts"]
-        snapshot_msg: OrderBookMessage = HuobiOrderBook.snapshot_message_from_exchange(
+    async def _order_book_snapshot(self, trading_pair: str) -> OrderBookMessage:
+        snapshot: Dict[str, Any] = await self.request_new_orderbook_snapshot(trading_pair)
+        snapshot_msg: OrderBookMessage = self.snapshot_message_from_exchange(
             msg=snapshot,
-            timestamp=timestamp,
             metadata={"trading_pair": trading_pair},
         )
-        order_book: OrderBook = self.order_book_create_function()
-        order_book.apply_snapshot(snapshot_msg.bids, snapshot_msg.asks, snapshot_msg.update_id)
-        return order_book
+
+        return snapshot_msg
 
     async def _subscribe_channels(self, ws: WSAssistant):
         try:
```

Comment:
I don't think we should use the trading pair as id of the message (lines 72 and 76). Also according to the documentation each message id should be unique, and we are using the same id for two different subscribe messages.